error: no rules expected the token `:`
 --> $DIR/invalid-bounds.rs:4:33
  |
4 |     struct Generics1<T: 'static : ?Sized> { //~ ERROR no rules expected the token `:`
  |                                 ^ no rules expected this token in macro call

error: no rules expected the token `:`
  --> $DIR/invalid-bounds.rs:10:33
   |
10 |     struct Generics2<T: 'static : Sized> { //~ ERROR no rules expected the token `:`
   |                                 ^ no rules expected this token in macro call

error: expected one of `+`, `,`, `=`, or `>`, found `:`
  --> $DIR/invalid-bounds.rs:15:1
   |
15 | / pin_project! {
16 | |     struct Generics3<T: Sized : 'static> { //~ ERROR expected one of `+`, `,`, `=`, or `>`, found `:`
17 | |         field: T,
18 | |     }
19 | | }
   | | ^
   | | |
   | |_expected one of `+`, `,`, `=`, or `>`
   |   in this macro invocation
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected one of `+`, `,`, `=`, or `>`, found `:`
  --> $DIR/invalid-bounds.rs:21:1
   |
21 | / pin_project! {
22 | |     struct Generics4<T: ?Sized : 'static> { //~ ERROR expected one of `+`, `,`, `=`, or `>`, found `:`
23 | |         field: T,
24 | |     }
25 | | }
   | | ^
   | | |
   | |_expected one of `+`, `,`, `=`, or `>`
   |   in this macro invocation
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected one of `+`, `,`, `=`, or `>`, found `:`
  --> $DIR/invalid-bounds.rs:27:1
   |
27 | / pin_project! {
28 | |     struct Generics5<T: Sized : ?Sized> { //~ ERROR expected one of `+`, `,`, `=`, or `>`, found `:`
29 | |         field: T,
30 | |     }
31 | | }
   | | ^
   | | |
   | |_expected one of `+`, `,`, `=`, or `>`
   |   in this macro invocation
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: no rules expected the token `Sized`
  --> $DIR/invalid-bounds.rs:34:34
   |
34 |     struct Generics6<T: ?Sized : Sized> { //~ ERROR no rules expected the token `Sized`
   |                                  ^^^^^ no rules expected this token in macro call

error: expected `where`, or `{` after struct name, found `Sized`
  --> $DIR/invalid-bounds.rs:39:1
   |
39 | / pin_project! {
40 | |     struct WhereClause1<T>
41 | |     where
42 | |         T: 'static Sized //~ ERROR expected `where`, or `{` after struct name, found `Sized`
...  |
45 | |     }
46 | | }
   | | ^
   | | |
   | |_expected `where`, or `{` after struct name
   |   in this macro invocation
   |
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: no rules expected the token `'static`
  --> $DIR/invalid-bounds.rs:51:18
   |
51 |         T: Sized 'static //~ ERROR no rules expected the token `'static`
   |                  ^^^^^^^ no rules expected this token in macro call
